BHoM_Engine/Query/IsNumeric.cs

```diff
@@ -47,59 +48,10 @@ public static bool IsNumeric(this Type type)
                 case TypeCode.Decimal:
                 case TypeCode.Double:
                 case TypeCode.Single:
-                    return true;
-                default:
-                    return false;
-            }
-        }
-
-        /***************************************************/
-
-        [Description("Determine whether a type is a floating-point numeric type." +
-            "\nSee https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/floating-point-numeric-types for more information.")]
-        [Input("type", "Type that we want to check if it is numeric type or not.")]
-        [Output("isNumeric", "True if the object is a numeric Type, false if not.")]
-        public static bool IsNumericFloatingPointType(this Type type)
-        {
-            // As per https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/floating-point-numeric-types
-
-            switch (Type.GetTypeCode(type))
-            {
-                case TypeCode.Decimal:
-                case TypeCode.Double:
-                case TypeCode.Single:
-                    return true;
-                default:
-                    return false;
-            }
-        }
-
-        /***************************************************/
-
-        [Description("Determine whether a type is a integral numeric type (an integer)." +
-            "\nSee https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/integral-numeric-types for more information.")]
-        [Input("type", "Type that we want to check if it is numeric type or not.")]
-        [Output("isNumeric", "True if the object is a numeric Type, false if not.")]
-        public static bool IsNumericIntegralType(this Type type)
-        {
-            // As per https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/integral-numeric-types
-
-            switch (Type.GetTypeCode(type))
-            {
-                case TypeCode.Byte:
-                case TypeCode.SByte:
-                case TypeCode.UInt16:
-                case TypeCode.UInt32:
-                case TypeCode.UInt64:
-                case TypeCode.Int16:
-                case TypeCode.Int32:
-                case TypeCode.Int64:
-                    return true;
+                    return !type.IsEnum || enumsAreNumbers;
```

Comment:
Could do this instead?
Guess functionality wise, will do the same, but could maybe be clearer (and doing exactly the same as the other two methods)
```
            switch (Type.GetTypeCode(type))
            {
                case TypeCode.Byte:
                case TypeCode.SByte:
                case TypeCode.UInt16:
                case TypeCode.UInt32:
                case TypeCode.UInt64:
                case TypeCode.Int16:
                case TypeCode.Int32:
                case TypeCode.Int64:
                    return !type.IsEnum || enumsAreNumbers;
                case TypeCode.Decimal:
                case TypeCode.Double:
                case TypeCode.Single:
                    return true;
                default:
                    return false;
            }

```